src/solvers/flattening/boolbv_index.cpp

```diff
@@ -132,7 +132,10 @@ bvt boolbvt::convert_index(const index_exprt &expr)
 
       // Simplify may remove the lower bound if the type
       // is correct.
-      prop.l_set_to_true(convert(simplify_expr(implication, ns)));
+      simplify(implication, ns);
+      const bvt &bv2 = convert_bv(implication);
+      CHECK_RETURN(bv2.size() == 1);
+      prop.l_set_to_true(bv2[0]);
```

Comment:
I recall changing uncached converts to cached ones here and it having a strange impact on performance.  It might be worth checking this.